aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -383,97 +392,175 @@ private Map<String, Parameter> getParametersToReset(
     ) {
-        return proxyClient.injectCredentialsAndInvokeV2(
-                Translator.describeEngineDefaultParametersRequest(model.getFamily(), parameterNames, marker),
-                proxyClient.client()::describeEngineDefaultParameters
-        );
+        Iterable<Parameter> iterable = Collections.emptyList();
+
+        if (filterParameterNames == null) {
+            final DescribeEngineDefaultParametersRequest request = DescribeEngineDefaultParametersRequest.builder()
+                    .dbParameterGroupFamily(dbParameterGroupFamily)
+                    .build();
+            iterable = fetchEngineDefaultParametersIterable(proxyClient, request);
+        } else {
+            for (final List<String> partition : Lists.partition(filterParameterNames, MAX_PARAMETER_FILTER_SIZE)) {
+                final Filter[] filters = new Filter[]{Translator.filterByParameterNames(partition)};
+                final DescribeEngineDefaultParametersRequest request = DescribeEngineDefaultParametersRequest.builder()
+                        .dbParameterGroupFamily(dbParameterGroupFamily)
+                        .filters(filters)
+                        .build();
+                iterable = Iterables.concat(iterable, fetchEngineDefaultParametersIterable(proxyClient, request));
+            }
+        }
+
+        return iterable;
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> describeEngineDefaultParameters(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final List<String> filterParameterNames,
+            final Map<String, Parameter> accumulator,
+            final RequestLogger logger
+    ) {
+        try {
+            final Iterable<Parameter> parameters = fetchEngineDefaultParametersIterableWithFilters(
+                    proxyClient,
+                    progress.getResourceModel().getFamily(),
+                    filterParameterNames
+            );
+            for (final Parameter parameter : parameters) {
+                accumulator.put(parameter.parameterName(), parameter);
+            }
+        } catch (Exception e) {
+            return Commons.handleException(progress, e, DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET);
+        }
+
+        return progress;
+    }
+
+    @VisibleForTesting
+    static Map<String, Parameter> computeModifiedDBParameters(
+            @NonNull final Map<String, Parameter> engineDefaultParameters,
+            @NonNull final Map<String, Parameter> currentDBParameters
+    ) {
+        final Map<String, Parameter> modifiedParameters = new HashMap<>();
+        for (final String paramName : currentDBParameters.keySet()) {
+            final Parameter currentParam = currentDBParameters.get(paramName);
+            final Parameter defaultParam = engineDefaultParameters.get(paramName);
```

Comment:
Can we calculate and log delta between `currentDBParameters` and `engineDefaultParameters`. Is happy scenario. It should be empty, However we saw some added attributes without being in Deafult parameters due to nature of engine. We need to have those explicitly in our logs if there any.